File: hdu/学_DP_导致的.cpp
#include <iostream>
#include <vector>
#include <set>
#include <algorithm>

using namespace std;

int inp_bignum(const string& x) {
    if (x.length() >= 3) {
        return 100;
    } else {
        return stoi(x);
    }
}

int main() {
    int t;
    cin >> t;
    while (t--) {
        string s, k_raw;
        cin >> s >> k_raw;
        int k = inp_bignum(k_raw);

        set<char> unique_chars(s.begin(), s.end());
        int ls = unique_chars.size();

        if (k >= ls) {
            cout << ls << endl;
        } else {
            string repeated_s;
            for (int i = 0; i < k; ++i) {
                repeated_s += s;
            }

            vector<int> dp(repeated_s.length(), 1);
            for (size_t i = 0; i < repeated_s.length(); ++i) {
                char vi = repeated_s[i];
                for (size_t j = 0; j < i; ++j) {
                    char vj = repeated_s[j];
                    if (vi > vj) {
                        dp[i] = max(dp[i], dp[j] + 1);
                    }
                }
            }

            int max_dp = *max_element(dp.begin(), dp.end());
            cout << max_dp << endl;
        }
    }
    return 0;
}

File: hdu/学_DP_导致的.py
def inp_bignum(x):
    if len(x) >= 3:
        return 100
    else:
        return int(x)


for _ in range(int(input())):
    s, k_raw = input().split()
    k = int(k_raw)

    ls = len(set(s))
    if k >= ls:
        print(ls)
    else:
        s = s * k
        dp = [1] * (len(s))
        for i in range(len(s)):
            vi = s[i]
            for j in range(i):
                j = j
                vj = s[j]
                if ord(vi) > ord(vj):
                    dp[i] = max(dp[i], dp[j] + 1)
    print(max(dp))
